src/plugins/proctracer/proctracer.cpp

```diff
@@ -0,0 +1,390 @@
+                        tracetrap->breakpoint.lookup_type = LOOKUP_NONE;
+                        tracetrap->breakpoint.addr_type = ADDR_PA;
+                        tracetrap->type = BREAKPOINT;
+                        tracetrap->name = "TraceTrap";
+                        tracetrap->cb = trace_cb;
+                        tracetrap->breakpoint.addr = trap_pa;
+                        tracetrap->data = ti;
+
+                        drakvuf_add_trap(drakvuf,tracetrap);
+                        p->trace_statistics[trap_pa]=0;
+                        mi->traps.push_back(tracetrap);
+                    }
+                    p->trace_status[cr3].push_back(mi);
+                }
+            }
+            vmi_free_unicode_str(us);
+        }
+
+        next_module = tmp_next;
+    }
+
+    return traps_ok;
+}
+
+// Runs on PsGetCurrentThreadTeb (process context)
+static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    proctracer *p = (proctracer*)info->trap->data;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    addr_t process = drakvuf_get_current_process(drakvuf, info->vcpu, info->regs);
+    char *proc_name = drakvuf_get_process_name(drakvuf, process);
+    switch(p->format){
+        case OUTPUT_CSV:
+            printf("proctracer,start,%lx,\"%s\"\n", info->regs->cr3, proc_name);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[PROCTRACER] Starting CR3: %lx NAME: %s\n", info->regs->cr3, proc_name);
+            break;
+    }
+    if (p->mod_config.find(proc_name) == p->mod_config.end() || p->trace_status.find(info->regs->cr3) != p->trace_status.end()){
+        free(proc_name);
+        drakvuf_release_vmi(drakvuf);
+        return 0;
+    }
+    add_trace_points(p, drakvuf, vmi, process, info->regs->cr3, proc_name);
+    free(proc_name);
+    drakvuf_release_vmi(drakvuf);
+    return 0;
+}
+
+proctracer::proctracer(drakvuf_t drakvuf, const void *config, output_format_t output) {
+    const struct proctracer_config *c = (const struct proctracer_config *)config;
+    const char *rekall_profile = c->rekall_profile;
+    const char *proctracer_config;
+    if (c->proctracer_config){
+        proctracer_config = c->proctracer_config;
+        printf("Proctracer config: %s\n",proctracer_config);
+    }else{
+        proctracer_config = "proctracer.json";    
+    }
+
+    printf("[PROCTRACER] Starting...\n");
+
+    this->offsets = (size_t*)g_malloc0(__OFFSET_MAX*sizeof(size_t));
+
+    for(int i=0;i<__OFFSET_MAX;i++) {
+        drakvuf_get_struct_member_rva(rekall_profile, offset_names[i][0], offset_names[i][1],
+                                      &this->offsets[i]);
+    }
+
+    json_object *conf_root = json_object_from_file(proctracer_config);
+    if (!conf_root){
+        printf("[PROCTRACER] Can't find config!\n");
+        return;
```

Comment:
I think all returns in the plugin constructor should actually throw an error instead of just returning. I see this is an issue with the other plugins too :/